plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/ElasticsearchServer.java

```diff
@@ -67,6 +68,7 @@ public ElasticsearchServer(Network network, String image)
         container = new ElasticsearchContainer(dockerImageName);
         container.withNetwork(network);
         container.withNetworkAliases("elasticsearch-server");
+        container.withStartupTimeout(Duration.ofMinutes(5));
```

Comment:
we should have `container.withStartupAttempts(3)` (instead??)

Reply:
That will make the total run time of the job longer, and it could still sometimes fail, if 3 attempts wouldn't be enough.

Additionally, testcontainers don't clean up failed containers, so we'd have to do it manually, if tests here have resource leak detection enabled.

Reply:
> That will make the total run time of the job longer

or shorter, depending on the mechanics
for example, if the container is doing something stupid which results in a much slower startup, aborting and trying again could lead to faster end to end

>  testcontainers don't clean up failed containers, so we'd have to do it manually, if tests here have resource leak detection enabled.

oh, so all the `withStartupAttempts` do not work today?
we should fix this in TC

Reply:
It doesn't work with https://github.com/trinodb/trino/blob/master/testing/trino-testing-containers/src/main/java/io/trino/testing/containers/junit/ReportLeakedContainers.java

> for example, if the container is doing something stupid which results in a much slower startup, aborting and trying again could lead to faster end to end

Have you observed something like this?

Reply:
> we should have `container.withStartupAttempts(3)`

`withStartupAttempts` is relevant when the container is flaky.

See https://github.com/testcontainers/testcontainers-java/blob/eadc9d6ec7777562d3b335b0e3ecd655e803e8c8/core/src/main/java/org/testcontainers/containers/GenericContainer.java#L1431-L1432


The problem at hand is not about the ES container sometimes failing to start because of internal issues, but rather the CI runner running ES7 & ES8 container may take some more time to spin both of the containers up while executing the tests in parallel.